cmd/bcpd/app/app_test.go

```diff
@@ -373,3 +436,103 @@ func queryAndCheckToken(t *testing.T, baseApp app.BaseApp, path string, data []b
 		require.Equal(t, expected[i], actual)
 	}
 }
+
+// makeSendTx is a special case of sendToken when the sender account is the only signer
+// this is used in our benchmark
+func makeSendTx(t require.TestingT, chainID string, sender, receiver *account, ticker, memo string, amount int64) []byte {
+	msg := &cash.SendMsg{
+		Src:  sender.address(),
+		Dest: receiver.address(),
+		Amount: &x.Coin{
+			Whole:  amount,
+			Ticker: ticker,
+		},
+		Memo: memo,
+	}
+
+	tx := &Tx{
+		Sum: &Tx_SendMsg{msg},
+	}
+
+	sig, err := sigs.SignTx(sender.pk, tx, chainID, sender.nonce())
+	require.NoError(t, err)
+	tx.Signatures = append(tx.Signatures, sig)
+	txBytes, err := tx.Marshal()
+	require.NoError(t, err)
+	require.NotEmpty(t, txBytes)
+	return txBytes
+}
+
+// benchmarkSendTx runs the actual benchmark sequence eg.
+// N * CheckTx
+// BeginBlock
+// N * DeliverTx
+// EndBlock
+// Commit
+func benchmarkSendTx(b *testing.B, nbAccounts, blockSize int) {
+	accounts := make([]*account, nbAccounts)
+	for i := 0; i < nbAccounts; i++ {
+		accounts[i] = &account{pk: crypto.GenPrivKeyEd25519()}
+	}
+
+	chainID := "bench-net-22"
+	myApp := newTestApp(b, chainID, accounts)
+
+	txs := make([][]byte, b.N)
+	for i := 1; i <= b.N; i++ {
+		sender := accounts[cmn.RandInt()%nbAccounts]
+		recipient := accounts[cmn.RandInt()%nbAccounts]
+		txs[i-1] = makeSendTx(b, chainID, sender, recipient, "ETH", "benchmark", 1)
+	}
+
+	b.ResetTimer()
+
+	for i := 1; i <= b.N; i++ {
```

Comment:
I would actually modify this a bit,
So we don't do all the check before all the deliver,
but process them in chunks of one block, with the checkTx right before delivertx

This doesn't matter right now but nice to make this a bit more realistic in another pr